### ISSUE TITLE
Source doesn't build nicely over debs.

### ISSUE THREAD

**Issue by aleeper** [moveit/moveit_ros#61]
If moveit debs are installed, some packages fail to build because they find the wrong headers. I believe this is caused by how the find_package and include_directories macros are used in the CMakeLists.txt files for various packages.

I believe the "best practices" for the CMake file is detailed here:
http://www.ros.org/wiki/catkin/CMakeLists.txt
and you will notice that things like include_directories should come late, if at all.
I think putting them before the catkin stuff is causing stuff to be found in the wrong order.

**Comment by isucan** [comment]
Can you try to move the includes lower and see if that fixes things?

**Comment by aleeper** [comment]
Yes, I just talked to William and got the rundown on what is the proper CMake style now. I'll start changing stuff.

**Comment by aleeper** [comment]
Ok I checked in changes to moveit_core, moveit_planners, and moveit_ros that seem to fix the compilation issues.

The main takeaways are: 
- use find_package() where possible.
- some packages (including Octomap...) do NOT follow convention and are inconsistent in the capitalization of the prefix they use in cmake, so they break the assumptions that catkin makes, so that the lib names can't be passed do downstream packages. In these cases, I used William and Dirk's suggestion to find the explicit lib path using cmake, and pass that to downstream packages through catkin_package().
- packages found through PkgConfig (e.g. fcl) can't be automatically passed to downstream packages, so I used the same explicit path trick to pass them through.
- add include directories after catkin_package()
- you don't need to use the SYSTEM keyword anywhere.
- you don't need to explicitly add link_directories for stuff that is a catkin package or that is found using "find_package"
- for targets that depend on messages, it is technically necessary to explicitly add a dependency like this: add_dependencies(${MOVEIT_LIB_NAME} moveit_msgs_gencpp), but we often get away without it when the build is fast, or single threaded, or isolated.
